core/bentley/src/test/OneAtATime.test.ts

```diff
@@ -2,51 +2,44 @@
+    const unhandledRejectionHandler = (_reason: any) => {
+      expect(_reason).toBeInstanceOf(Error);
+    };
+
+    process.on("unhandledRejection", unhandledRejectionHandler);
     let calls = 0;
     const operation = new OneAtATimeAction(async (a: number, b: string) => {
       if (a === 10)
         throw new Error("cancelled");
 
-      assert.equal(a, 200);
-      assert.equal(b, "hello");
+      expect(a).toBe(200);
+      expect(b).toBe("hello");
       await BeDuration.wait(100);
       return ++calls;
     }, "testAbandon");
 
-    // expect(operation.request(200, "hello")).to.be.eventually.fulfilled; // is started immediately
-    // expect(operation.request(200, "hello")). ; // is started immediately
-    // operation.request(200, "hello");
     const abandonedError = new AbandonedError("testAbandon");
     const cancelled = new AbandonedError("cancelled");
 
-    expect(operation.request(200, "hello")).rejects.with.toBeInstanceOf(AbandonedError); // becomes pending, doesn't abort previous because its already started
+    void expect(operation.request(200, "hello")).rejects.with.toBeInstanceOf(AbandonedError); // becomes pending, doesn't abort previous because its already started
 
-    expect(operation.request(200, "hello")).rejects.with.toBeInstanceOf(AbandonedError); // aborts previous, becomes pending
+    void expect(operation.request(200, "hello")).rejects.with.toBeInstanceOf(AbandonedError); // aborts previous, becomes pending
     let count = await operation.request(200, "hello"); // aborts previous, becomes pending, eventually is run
-    assert.equal(count, 2); // only the first and last complete
-    // console.log(await operation.request(10, "hello"));
+    expect(count).toBe(2); // only the first and last complete
     // then, just try the whole thing again
-    expect(operation.request(10, "hello")).rejects; // try calling a function that throws
-    expect(operation.request(10, "hello")).rejects.with.toEqual(cancelled); // try calling a function that throws
-    expect(operation.request(200, "hello")).rejects.with.toEqual(abandonedError); // becomes pending, doesn't abort previous because its already started
-    expect(operation.request(200, "hello")).rejects.with.toEqual(abandonedError); // aborts previous, becomes pending
+    void expect(operation.request(10, "hello")).rejects; // try calling a function that throws
+    void expect(operation.request(10, "hello")).rejects.with.toEqual(cancelled); // try calling a function that throws
+    void expect(operation.request(200, "hello")).rejects.with.toEqual(abandonedError); // becomes pending, doesn't abort previous because its already started
+    void expect(operation.request(200, "hello")).rejects.with.toEqual(abandonedError); // aborts previous, becomes pending
     count = await operation.request(200, "hello");
-    assert.equal(count, 3);
+    expect(count).toBe(3);
+    process.removeListener("unhandledRejection", unhandledRejectionHandler);
```

Comment:
What happens if the test exits early?

Reply:
yeah, to ensure the test does not exit early, I added a [expect.assertions](https://vitest.dev/api/expect#expect-assertions) and if the test exits early, that assertion would fail

Reply:
Will that ensure the listener is removed, though? 
It also requires us to track expectations should we add more... what about [onTestFinished](https://vitest.dev/api/#ontestfinished)?

Reply:
that looks good, okay, I added that

Reply:
Sorry, will it work at the bottom of the test? Should it be moved up?

Reply:
good point, I've moved it up

Reply:
thanks